subsys/bluetooth/mesh/dfd_srv.c

```diff
@@ -925,12 +925,35 @@ const struct bt_mesh_blob_srv_cb _bt_mesh_dfd_srv_blob_cb = {
 
 static int dfd_srv_init(const struct bt_mesh_model *mod)
 {
+	int err;
 	struct bt_mesh_dfd_srv *srv = mod->rt->user_data;
 
 	srv->mod = mod;
 
-	if (IS_ENABLED(CONFIG_BT_MESH_MODEL_EXTENSIONS)) {
-		bt_mesh_model_extend(mod, srv->upload.blob.mod);
+	const struct bt_mesh_model *blob_srv =
+		bt_mesh_model_find(bt_mesh_model_elem(mod), BT_MESH_MODEL_ID_BLOB_SRV);
+
+	if (blob_srv == NULL) {
+		LOG_ERR("Firmware Distribution Server cannot extend BLOB Transfer Server");
```

Comment:
Due to how we organize and discover models in the NCS Mesh, and how specification indicates they should arranged, I think it is better to shorten and reword this as:

```suggestion
		LOG_ERR("BLOB Transfer srv can't be on same element");
```

Long Debug string are not very useful and eat up the flash during development stage leading to misconceptions about footprint.

Reply:
Here the check is to make sure that Firmware Distribution Server is at the same element with BLOB Transfer server. Quite opposite of what you suggesting! Would you like me to update the commit and PR message if it is not clear enough?